File: Source/JapaneseLanguageTools.Contracts/Models/Requests/SubmitFailedCharacterExerciseDetailsRequest.Model.cs
using System.Collections.Generic;

using JapaneseLanguageTools.Contracts.Models.Responses;

namespace JapaneseLanguageTools.Contracts.Models.Requests;

public class SubmitFailedCharacterExerciseDetailsRequestModel
{
    public IList<CharacterExerciseModel> FailedItems { get; set; } = [];

    public required GenerateCharacterExerciseBatchResponseModel OriginalResponse { get; set; }
}

File: Source/JapaneseLanguageTools.ReactSpa/source/models/requests/SubmitFailedCharacterExerciseDetailsRequest.Model.ts
import CharacterExerciseModel from "@/models/CharacterExercise.Model";
import GenerateCharacterExerciseBatchResponseModel from "@/models/responses/GenerateCharacterExerciseBatchResponse.Model";

export default interface SubmitFailedCharacterExerciseDetailsRequestModel {
  failedItems: CharacterExerciseModel[];

  originalResponse: GenerateCharacterExerciseBatchResponseModel;
}
